### ISSUE TITLE
Disk space not freed up from cache

### ISSUE THREAD

**Issue by meszape** [astrada/google-drive-ocamlfuse#512]
Hi

I ran into an issue, that my disk became 100% filled, however analyzing my filesystem I only used 20%. I did some digging and found, that when files are deleted from cache, the data doesn't gets deleted, becasue the files are still open.

I double checked and all uploads finished. I closed the application which was writting to googledrive, waited a few hours, but the space is still occupied, and i can see the cache files if i execute
`sudo lsof | grep 'delete'`

So to sum up:

- My cache directory is empty

- `sudo lsof | grep 'deleted'` shows that 20GB cache files are deleted, but still there

- Verified that modified files are updated in googledrive

- Verified that there is no network activity

- Verified with `df -h` that 20GB of space is still filled

Please advise!

Thanks!

**Comment by astrada** [comment]
Should be fixed in the beta [ppa](https://launchpad.net/~alessandro-strada/+archive/ubuntu/google-drive-ocamlfuse-beta). If you are using opam, you can try the fix before it gets to the main repo, running:

    opam pin -n add gapi-ocaml https://github.com/astrada/gapi-ocaml#master
    opam update
    opam upgrade

To restore the main repo version:

    opam pin -n remove gapi-ocaml
    opam update
    opam upgrade

Thanks for reporting the issue!

**Comment by Calendal** [comment]
> Should be fixed in the beta [ppa](https://launchpad.net/~alessandro-strada/+archive/ubuntu/google-drive-ocamlfuse-beta). If you are using opam, you can try the fix before it gets to the main repo, running:
> 
> ```
> opam pin -n add gapi-ocaml https://github.com/astrada/gapi-ocaml#master
> opam update
> opam upgrade
> ```
> To restore the main repo version:
> 
> ```
> opam pin -n remove gapi-ocaml
> opam update
> opam upgrade
> ```
> Thanks for reporting the issue!

This is a fix or a workarround?

**Comment by astrada** [comment]
> This is a fix or a workarround?

It should fix the issue. The `opam pin` commands are to test the fix before it gets approved and lands in the official opam-repository.

**Comment by Calendal** [comment]
> > This is a fix or a workarround?
> 
> It should fix the issue. The `opam pin` commands are to test the fix before it gets approved and lands in the official opam-repository.

OK! Thanks for the effort.

**Comment by meszape** [comment]
I can confirm, that this is fixed!
Thx
